src/main/java/org/jenkinsci/plugin/Diskcheck.java

```diff
@@ -10,195 +11,180 @@
 import hudson.tasks.BuildWrapper;
 import hudson.node_monitors.*;
 import hudson.model.Node;
-import hudson.model.Result;
-import hudson.tasks.BatchFile;
-import hudson.tasks.CommandInterpreter;
-import hudson.tasks.Shell;
 
 import org.kohsuke.stapler.DataBoundConstructor;
 
-import java.io.File;
 import java.io.IOException;
 import java.io.PrintStream;
-
-
+import java.net.UnknownHostException;
+import java.util.List;
 
 /**
  * Class to allow any build step to be performed before the SCM checkout occurs.
- * 
+ *
  * @author Manoj Thakkar
+ * @author Frano Stanisic (8th Decemeber 2015, updated deletion to use Jenkins API/only delete files up to a threshhold)
  * 
  */
+public class Diskcheck extends BuildWrapper {
+
+    public final boolean failOnError;
+
+    public static FilePath lastFileModified(PrintStream log, FilePath dir) throws UnknownHostException, IOException, InterruptedException {
+        log.println("Determining file to delete ...");
+        List<FilePath> files = dir.listDirectories();
+        long lastMod = Long.MAX_VALUE;
+        FilePath choice = null;
+        for (FilePath file : files) {
+            if (file.lastModified() < lastMod) {
+                choice = file;
+                lastMod = file.lastModified();
+            }
+        }
+        return choice;
+    }
+
+    /**
+     * Constructor taking a list of buildsteps to use.
+     *
+     * @param buildstep list of build steps configured in the UI
+     */
+    @DataBoundConstructor
+    public Diskcheck(boolean failOnError) {
+        this.failOnError = failOnError;
+    }
+
+    /**
+     * Overridden setup returns a noop class as we don't want to add anything
+     * here.
+     *
+     * @param build
+     * @param launcher
+     * @param listener
```

Comment:
Please don't create Javadoc entries without descriptions. When you try to release the plugin on JDK8, it blows up